File: MeoVat/Components/TipPhoto.swift
import SwiftUI

/// Photo from the API with the bundled asset of the same name as placeholder / offline fallback.
struct TipPhoto: View {
    let url: String?
    let fallback: String

    var body: some View {
        AppColor.canvasSecondary
            .overlay {
                if let url, let remote = URL(string: url) {
                    AsyncImage(url: remote) { phase in
                        if let image = phase.image {
                            image.resizable().scaledToFill()
                        } else {
                            fallbackImage
                        }
                    }
                } else {
                    fallbackImage
                }
            }
            .clipped()
    }

    @ViewBuilder
    private var fallbackImage: some View {
        if UIImage(named: fallback) != nil {
            Image(fallback).resizable().scaledToFill()
        }
    }
}

File: app/src/main/java/com/fatties/meovat/ui/components/TipPhoto.kt
package com.fatties.meovat.ui.components

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import coil3.compose.AsyncImage
import com.fatties.meovat.data.Photos
import com.fatties.meovat.ui.theme.AppColor

/** Photo from the API with the bundled drawable of the same name as placeholder / offline fallback. */
@Composable
fun TipPhoto(url: String?, fallback: String?, modifier: Modifier = Modifier) {
    val fallbackId = fallback?.let(Photos::idOrNull)
    val fallbackPainter = fallbackId?.let { painterResource(it) }
    Box(modifier.background(AppColor.CanvasSecondary)) {
        if (url != null) {
            AsyncImage(
                model = url,
                contentDescription = null,
                contentScale = ContentScale.Crop,
                placeholder = fallbackPainter,
                error = fallbackPainter,
                modifier = Modifier.matchParentSize(),
            )
        } else if (fallbackPainter != null) {
            Image(painter = fallbackPainter, contentDescription = null, contentScale = ContentScale.Crop, modifier = Modifier.matchParentSize())
        }
    }
}
